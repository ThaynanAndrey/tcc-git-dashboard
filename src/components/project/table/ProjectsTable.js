import React from 'react';
import StripedTable from '../../layout/StripedTable';
import TableHead from '../../layout/TableHead';
import ProjectItensTable from './ProjectItensTable';

/**
 * Stateless component with the function to presenting the New Project Modal.
 * 
 * @author Thaynan Nunes
 */
const ProjectsTable = ({ projects, openProject, deleteProject }) => {
    return (
        projects.length > 0 && 
        <StripedTable>
            <TableHead headsNames={["Nome", "Data de Criação", ""]}/>
            <ProjectItensTable
                projects={projects}
                openProject={openProject}
                deleteProject={deleteProject} />
        </StripedTable>
    );
}

export default ProjectsTable;