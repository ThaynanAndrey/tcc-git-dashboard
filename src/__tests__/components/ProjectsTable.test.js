import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import ProjectsTable from '../../components/project/table/ProjectsTable';

describe('Test ProjectsTable', () => {
  let tree, projects, openProject, deleteProject;
  
  beforeEach(() => {
    projects = [{
        name: 'test',
        creationDate: '10/10/1010'
    }];
    openProject = jest.fn();
    deleteProject = jest.fn();

    tree = mount(
        <ProjectsTable
            projects={projects}
            openProject={openProject}
            deleteProject={deleteProject}
        />
    );
  });

  it('renders correctly a component to show ProjectsTable', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a component to show ProjectsTable with correct props', () => {
    expect(tree.prop('projects')).toEqual(projects);
    expect(tree.prop('openProject')).toEqual(openProject);
    expect(tree.prop('deleteProject')).toEqual(deleteProject);
  });
});