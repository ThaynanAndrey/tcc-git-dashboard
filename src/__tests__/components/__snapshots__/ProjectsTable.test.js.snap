// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProjectsTable renders correctly a component to show ProjectsTable 1`] = `
<ProjectsTable
  deleteProject={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "creationDate": "10/10/1010",
            "name": "test",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  openProject={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "creationDate": "10/10/1010",
            "name": "test",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  projects={
    Array [
      Object {
        "creationDate": "10/10/1010",
        "name": "test",
      },
    ]
  }
>
  <StripedTable>
    <table
      className="striped highlight responsive-table"
    >
      <TableHead
        headsNames={
          Array [
            "Nome",
            "Data de Criação",
            "",
          ]
        }
      >
        <thead>
          <tr>
            <th
              key="0"
            >
              Nome
            </th>
            <th
              key="1"
            >
              Data de Criação
            </th>
            <th
              key="2"
            />
          </tr>
        </thead>
      </TableHead>
      <ProjectItensTable
        deleteProject={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "creationDate": "10/10/1010",
                  "name": "test",
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        openProject={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "creationDate": "10/10/1010",
                  "name": "test",
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        projects={
          Array [
            Object {
              "creationDate": "10/10/1010",
              "name": "test",
            },
          ]
        }
      >
        <tbody>
          <tr
            key="0"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <td>
              test
            </td>
            <td>
              10/10/1010
            </td>
            <td>
              <Tooltip.Wrapper
                arrow={8}
                background="#000"
                border="#000"
                color="#fff"
                content="Remover Projeto"
                fadeDuration={0}
                fadeEasing="linear"
                fixed={false}
                fontFamily="inherit"
                fontSize="inherit"
                offset={0}
                padding={16}
                placement="left"
                radius={10}
                style={
                  Object {
                    "whiteSpace": "nowrap",
                  }
                }
                zIndex={1}
              >
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                  css={
                    Object {
                      "map": undefined,
                      "name": "1oin9j2",
                      "next": undefined,
                      "styles": "
          
  position: relative;
  display: inline-block;
;
          
        ",
                    }
                  }
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <div
                    className="css-1oin9j2"
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "whiteSpace": "nowrap",
                      }
                    }
                  >
                    <i
                      className="material-icons left delete-text red-text"
                      style={
                        Object {
                          "cursor": "pointer",
                        }
                      }
                    >
                      delete
                    </i>
                    <Tooltip
                      fadeDuration={0}
                      fadeEasing="linear"
                      offset={8}
                      open={false}
                      placement="left"
                      zIndex={1}
                    />
                  </div>
                </EmotionCssPropInternal>
              </Tooltip.Wrapper>
            </td>
          </tr>
        </tbody>
      </ProjectItensTable>
    </table>
  </StripedTable>
</ProjectsTable>
`;
